src/cargo/core/compiler/custom_build.rs

```diff
@@ -884,10 +957,16 @@ impl BuildOutput {
                 "rustc-flags" => {
                     let (paths, links) = BuildOutput::parse_rustc_flags(&value, &whence)?;
                     library_links.extend(links.into_iter());
-                    library_paths.extend(paths.into_iter());
+                    library_paths.extend(paths.into_iter().map(|p| {
+                        LibraryPath::new(p, script_out_dir, script_out_dir_when_generated)
+                    }));
                 }
                 "rustc-link-lib" => library_links.push(value.to_string()),
-                "rustc-link-search" => library_paths.push(PathBuf::from(value)),
+                "rustc-link-search" => library_paths.push(LibraryPath::new(
+                    PathBuf::from(value),
+                    script_out_dir,
+                    script_out_dir_when_generated,
```

Comment:
Why do we still need `script_out_dir_when_generated` here, isn't `value` already rewritten with `script_out_dir` at line 932?

Reply:
I was being overly defensive. It should just pass in script_out_dir.